src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1759,6 +1771,14 @@
                    :listeners (get options :listeners  [])
                    :get-alphas-fn get-alphas-fn})))
 
+(defn add-production-load-order
+  "Adds ::rule-load-order to metadata of productions. Custom DSL's may need to use this if
+   creating a session in Clojure without calling mk-session below."
+  [productions]
+  (map (fn [n production]
+         (vary-meta production assoc ::rule-load-order (or n 0)))
+       (range) productions))
```

Comment:
I'm hesitant to expand the surface area of functions that users would call directly in the compiler namespace, since most of these functions are very much internal implementation details of Clara and should really be considered "project private".  Perhaps we could add this in the DSL namespace or somewhere else instead?

Thoughts anyone?

Reply:
It seems like perhaps there'd be benefit of having a `clara.internal.*` ns prefix if the number of these sorts of things grew. That'd at least make it more clear where the boundaries are.

In Leiningen, there is an `:internal` metadata key added to indicate these sorts of situations. That seems reasonable to me, but would be something to document since there are no enforcement around that sort of thing. It's just a convention.

Reply:
Hmm.. so on reflection having the compiler require the DSL namespace seems odd to me.  We can probably just leave this in the compiler for the moment.  It is certainly not the only place where the boundaries between public APIs and implementation details are somewhat ambiguous, particularly since it isn't necessarily just a case of labeling functions public vs private.  A major discussion around that is out of scope here IMO so we can just set it aside for the time being.  We might want to move this function later, but if the change would just be to require the same function from another namespace that probably wouldn't be too disruptive to the probably-limited pool of users that would call this function.

Reply:
>  so on reflection having the compiler require the DSL namespace seems odd to me

Yeah, I think that sounds wrong.